independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/BeanDeployment.java

```diff
@@ -1087,6 +1089,29 @@ private List<BeanInfo> findBeans(Collection<DotName> beanDefiningAnnotations, Li
                 beans.add(classBean);
                 beanClassToBean.put(beanClass, classBean);
                 injectionPoints.addAll(classBean.getAllInjectionPoints());
+
+                // specification requires to disallow non-static public fields on non-`@Dependent` beans,
+                // but we do what Weld does: only disallow them on normal scoped beans (disallowing them
+                // on `@Singleton` beans would actually prevent passing the AtInject TCK)
+                //
+                // only check this in the strictly compatible mode, as this is pretty invasive,
+                // it even prevents public producer fields
+                if (classBean.getScope().isNormal() && strictCompatibility) {
```

Comment:
This is an example of  a validation that IMO makes perfect sense even without the `strictCompatibility=true`.

Reply:
I would have agreed, but we have a few tests in ArC test suite that fail with this check (notably, `public` producer field on a normal-scoped bean -- which seems fine to me?), which suggests to me that it's probably more invasive than we'd like.

Reply:
I'd keep it under strict check, I don't even see that much value in this check to begin with.

Reply:
> notably, `public` producer field on a normal-scoped bean -- which seems fine to me?

Hmm, I see. Let's keep it in the strict mode then.